master/usage/encrypt-comms.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Configuring encryption and authentication
+canonical_url: 'https://docs.projectcalico.org/master/usage/encrypt-comms'
+---
+
+## etcd connections
+
+If you are using the etcd datastore, we recommend enabling mutual TLS authentication on
+its connections as follows.
+
+- [Configure etcd](https://coreos.com/etcd/docs/latest/op-guide/security.html) to encrypt its
+  communications with TLS and require clients to present certificates signed by the etcd certificate
+  authority.
+
+- Configure each {{site.prodname}} component to verify the etcd server's identity and to present
+  a certificate to the etcd server that is signed by the etcd certificate authority.
+  - [{{site.nodecontainer}}](../reference/node/configuration)
+  - [`calicoctl`](./calicoctl/configure/etcd)
+  - [CNI plugin](../reference/cni-plugin/configuration#etcd-location)
+  - [Felix](../reference/felix/configuration#etcdv3-datastore-configuration) (on [bare metal hosts](../getting-started/bare-metal/installation/))
+
+### kube-apiserver communications (Kubernetes and OpenShift)
+
+All communications with kube-apiserver occur over TLS 1.2 with client authentication by
+default. The {{site.prodname}} components authenticate to kube-apiserver with either an x.509 certificate
+or a [JSON web token (JWT)](https://jwt.io/). You do not need to take action to secure these
+communications.
+
+### Typha connections (Kubernetes)
+
+{% include {{page.version}}/felix-typha-tls-intro.md %}
+
+To use TLS, each Typha instance must have a certificate and key pair signed by
+a trusted CA.  Typha then only accepts TLS connections, and requires each
+connecting client to present a certificate that is signed by a trusted CA and
+has an expected identity in its Common Name or URI SAN field.  Either
+`ClientCN` or `ClientURISAN` must be configured, and Typha will check the
+presented certificate accordingly.
+
+-  For a [SPIFFE](https://github.com/spiffe/spiffe)-compliant deployment you
+   should configure `ClientURISAN` with a [SPIFFE
+   Identity](https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#2-spiffe-identity)
+   and provision client certificates with the same identity in their URI SAN
+   field.
+
+-  Alternatively you can configure `ClientCN` and provision client certificates
+   with that `ClientCN` value in their Common Name field.
+
+If both those parameters are set, a client certificate only has to match one of
+them; this is intended for when a deployment is migrating from using Common
+Name to using a URI SAN, to express identity.
+
+| Configuration parameter | Environment variable   | Description | Schema |
```

Comment:
This table should stay in reference, and we should link to it from here.

Reply:
We need to talk about what's needed for the Felix end of the connection as well.  Which means that we need to merge in the relevant text from .../reference/felix/configuration

Reply:
Oh dang, missed that somehow. Good point.